internal/api/token_refresh.go

```diff
@@ -36,111 +37,152 @@ func (a *API) RefreshTokenGrant(ctx context.Context, w http.ResponseWriter, r *h
 		return oauthError("invalid_request", "refresh_token required")
 	}
 
-	user, _, session, err := models.FindUserWithRefreshToken(db, params.RefreshToken, false)
-	if err != nil {
-		if models.IsNotFoundError(err) {
-			return oauthError("invalid_grant", "Invalid Refresh Token: Refresh Token Not Found")
-		}
-		return internalServerError(err.Error())
-	}
-
-	if user.IsBanned() {
-		return oauthError("invalid_grant", "Invalid Refresh Token: User Banned")
-	}
+	// A 5 second retry loop is used to make sure that refresh token
+	// requests do not waste database connections waiting for each other.
+	// Instead of waiting at the database level, they're waiting at the API
+	// level instead and retry to refresh the locked row every 10-30
+	// milliseconds.
+	retryStart := time.Now()
+	retry := true
 
-	if session != nil {
-		var notAfter time.Time
+	for retry && time.Since(retryStart).Seconds() < 5.0 {
```

Comment:
/nit aware it's in the comment above but can we convert the retry loop duration into a const somewhere

Reply:
What do you mean? In a "const?"